bounceview/src/main/java/com/github/hariprasanths/bounceview/CustomDialog.java

```diff
@@ -0,0 +1,243 @@
+                break;
+            case 1 :
+                button = (Button) baseButtonLayout.findComponentById(ResourceTable.Id_bb1);
+                break;
+            case 2:
+                button = (Button) baseButtonLayout.findComponentById(ResourceTable.Id_bb2);
+                break;
+            default :
+                return this;
+        }
+        button.setVisibility(0);
+        button.setText(text);
+        button.setMarginsLeftAndRight(leftMargin, rightMargin);
+        button.setClickedListener(listener);
+        this.setButtonCustomComponent(baseButtonLayout);
+        return this;
+    }
+
+    /**
+     * Does some thing in old style.
+     * use {@link #setCommonImageButton(int, int, int, int, Component.ClickedListener)} instead.
+     */
+    @Override
+    public CommonDialog setImageButton(int buttonNum, int resId, ClickedListener listener)
+            throws UnsupportedOperationException {
+        String message = "setButton is unsupported, use setCommonImageButton(int buttonNum, int resId, int leftMargin,"
+                + " int rightMargin, Component.ClickedListener listener) instead.";
+        throw new UnsupportedOperationException(message);
+    }
+
+    /**
+     * set Image buttons for the custom common dialog.
+     *
+     * @param buttonNum by default at most 3 buttons can be added, takes 0,1,2.
+     * @param resId ResourceId for Image that you want to set
+     * @param leftMargin left margin for image button
+     * @param rightMargin right margin for image button
+     * @param listener component listener
+     * @return Custom common dialog
+     */
+    public CommonDialog setCommonImageButton(int buttonNum, int resId, int leftMargin, int rightMargin,
+                                             Component.ClickedListener listener) {
+        if (mButtonComponent != null) {
+            return this;
+        }
+        Image image;
+        switch (buttonNum) {
+            case 0 :
+                image = (Image) baseImageButtonLayout.findComponentById(ResourceTable.Id_imgb0);
+                break;
+            case 1 :
+                image = (Image) baseImageButtonLayout.findComponentById(ResourceTable.Id_imgb1);
+                break;
+            case 2 :
+                image = (Image) baseImageButtonLayout.findComponentById(ResourceTable.Id_imgb2);
+                break;
+            default:
+                return this;
+        }
+        image.setVisibility(0);
+        image.setMarginsLeftAndRight(leftMargin, rightMargin);
+        image.setPixelMap(resId);
+        image.setClickedListener(listener);
+        mImageButtonComponent = baseImageButtonLayout;
+        return this;
+    }
+
+    @Override
+    public void show() {
+        DirectionalLayout componentContainer = new DirectionalLayout(context);
+        int k = 0;
+        if (mTitleComponent != null) {
+            componentContainer.addComponent(mTitleComponent, k);
+            k++;
```

Comment:
perform index++ in the penultimate statement

Reply:
done